src/main/flight/gps_rescue.c

```diff
@@ -669,25 +672,28 @@ void descend(void)
         rescueState.intent.targetVelocityCmS = gpsRescueConfig()->rescueGroundspeed * rescueState.intent.proximityToLandingArea;
         // reduce target velocity as we get closer to home. Zero within 2m of home, reducing risk of overshooting.
         // if quad drifts further than 2m away from home, should by then have rotated towards home, so pitch is allowed
-        rescueState.intent.rollAngleLimitDeg = 0.5f * gpsRescueConfig()->maxRescueAngle * rescueState.intent.proximityToLandingArea;
-        // reduce roll capability when closer to home, none within final 2m
-        rescueState.intent.pitchAngleLimitDeg = gpsRescueConfig()->maxRescueAngle * rescueState.sensor.groundspeedPitchAttenuator;
+        rescueState.intent.rollAngleLimitDeg = 0.5f * rescueState.intent.pitchAngleLimitDeg * (rescueState.sensor.distanceToHomeM / rescueState.intent.descentDistanceM);
+        // reduce roll capability when closer to home, keeping a little bit within final 2m to prevent endless yaw spin at 2m
+        // limit roll angle from GPS Rescue code to half the max pitch angle; earth referenced yaw may add more roll via angle code
     }
 
     rescueState.intent.yawAttenuator = 1.0f; // just in case entered descend phase before yaw authority was complete
 
-    // configure altitude step for descent, considering interval between altitude readings
-    rescueState.intent.altitudeStep = -1.0f * rescueState.sensor.altitudeDataIntervalSeconds * gpsRescueConfig()->descendRate;
+    rescueState.intent.altitudeStep = rescueState.sensor.altitudeDataIntervalSeconds * gpsRescueConfig()->descendRate;
```

Comment:
```suggestion
    float altitudeStep *= rescueState.sensor.altitudeDataIntervalSeconds * gpsRescueConfig()->descendRate;
    .....
    rescueState.intent.altitudeStep = altitudeStep;
```

But ti may be better to calculate/use `rescueState.intent.verticalSpeed` and multiply it by altitudeDataIntervalSeconds  just before applying it.

And `altitudeStep` is always computed locally. Is is really needed to sore it in state (except for debugging)?

Reply:
Yes I can use altitude step locally, I'll do that.